website/addons/owncloud/views.py

```diff
@@ -0,0 +1,134 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from furl import furl
+
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+import owncloud
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+from website.util import api_url_for
+
+from website.addons.owncloud import settings
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_root_folder = generic_views.root_folder(
+    SHORT_NAME
+)
+
+@must_be_logged_in
+def owncloud_user_config_get(auth, **kwargs):
+    """View for getting a JSON representation of the logged-in user's
+    OwnCloud user settings.
+    """
+
+    user_addon = auth.user.get_addon('owncloud')
+    user_has_auth = False
+    if user_addon:
+        user_has_auth = user_addon.has_auth
+
+    return {
+        'result': {
+            'userHasAuth': user_has_auth,
+            'urls': {
+                'create': api_url_for('owncloud_add_user_account'),
+                'accounts': api_url_for('owncloud_account_list'),
+            },
+            'hosts': DEFAULT_HOSTS,
+        },
+    }, http.OK
+
+
+## Config ##
+
+@must_be_logged_in
+def owncloud_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+
+    # Ensure that ownCloud uses https
+    host = furl(request.json.get('host').rstrip('/'))
+    host.scheme = 'https'
```

Comment:
Some elaboration on this: it was a url of the form `subdomain.domain.tld`. `furl` seemed, for whatever reason, to this that this was a path when it was instantiated, rather than a host. The following snippet worked:
```
host_url = request.json.get('host').strip('/')
host = furl()
host.host = host_url
host.scheme = 'https'
```